core/test/XiosWrite_test.cpp

```diff
@@ -102,37 +105,47 @@ MPI_TEST_CASE("TestXiosWrite", 2)
 
     xios_handler.close_context_definition();
 
-    // --- Tests for file API
-    // create some fake data to test writing methods
-    double* field_2D = new double[n1 * n2];
-    double* field_3D = new double[n1 * n2 * n3];
-    double* field_4D = new double[n1 * n2 * n3 * n4];
-    for (size_t idx = 0; idx < n1 * n2; idx++) {
-        field_2D[idx] = 1.0 * idx;
+    // --- Tests for writing to file
+    Module::setImplementation<IStructure>("Nextsim::ParametricGrid");
+    ModelArray::setDimension(ModelArray::Dimension::X, n1);
+    ModelArray::setDimension(ModelArray::Dimension::Y, n2);
+    ModelArray::setDimension(ModelArray::Dimension::Z, n3);
+    // Create some fake data to test writing methods
+    HField field_2D(ModelArray::Type::H);
+    field_2D.resize();
+    for (size_t j = 0; j < n2; ++j) {
+        for (size_t i = 0; i < n1; ++i) {
+            field_2D(i, j) = 1.0 * (i + n1 * j);
+        }
     }
-    for (size_t idx = 0; idx < n1 * n2 * n3; idx++) {
-        field_3D[idx] = 1.0 * idx;
-    }
-    for (size_t idx = 0; idx < n1 * n2 * n3 * n4; idx++) {
-        field_4D[idx] = 1.0 * idx;
+    HField field_3D(ModelArray::Type::Z);
+    field_3D.resize();
+    for (size_t k = 0; k < n3; ++k) {
+        for (size_t j = 0; j < n2; ++j) {
+            for (size_t i = 0; i < n1; ++i) {
+                field_3D(i, j, k) = 1.0 * (i + n1 * (j + n2 * k));
+            }
+        }
     }
+    // TODO: field_4D?
```

Comment:
I'd taken a note that we should support I/O 2D, 3D, and 4D fields. However, looking at https://github.com/nextsimhub/nextsimdg/blob/main/core/src/discontinuousgalerkin/ModelArrayDetails.cpp I only see `ModelArray`s with 2 or 3 components. @einola @timspainNERSC please could you remind me what the use case is for 4D?